src/key_value_store/mod.rs

```diff
@@ -0,0 +1,51 @@
+use core::cell::Cell;
+
+// fixme this should probably go elsewhere
+pub trait SafelyDeserializable {
+    fn can_deserialize_from(bytes: &[Cell<u8>]) -> bool;
+}
+
+impl SafelyDeserializable for u8 {
+    fn can_deserialize_from(_bytes: &[Cell<u8>]) -> bool {
+        true
+    }
+}
+
+impl SafelyDeserializable for u16 {
+    fn can_deserialize_from(_bytes: &[Cell<u8>]) -> bool {
+        true
```

Comment:
Shouldn't this test that the `_bytes` is large enough for the type?

Reply:
Probably, but the key value store already does that check. I guess it depends upon how tightly we want to tie this type to `KeyValueStore`.

Reply:
Who would implement this as false then?

Reply:
@CollinDietz you'd need to provide a "real" implementation for at least some user-defined types. For example, let's say you have an enum that can only take certain values. You'd need an implementation that can check for those values before saying that you can deserialize from the provided bytes.

I'm hoping that we only have to do this rarely and can derive `SafelyDeserializable` this from most aggregate types.